nx.json

```diff
@@ -3,11 +3,24 @@
     "default": {
       "runner": "nx/tasks-runners/default",
       "options": {
-        "cacheableOperations": ["test:lib"]
+        "cacheableOperations": [
+          "test:lib",
+          "test:eslint",
+          "test:types",
+          "build:types",
+          "test:format",
+          "test:build",
+          "build"
+        ]
       }
     }
   },
   "defaultBase": "main",
+  "pluginsConfig": {
+    "@nrwl/js": {
+      "analyzeSourceFiles": false
+    }
+  },
```

Comment:
*short explanation*: this exlicitely states our depency graph strategy - which is to analyze deps based on package.json files. This was the behavior we are already using.

*long explanation*: I added the `@nrwl/workspace` briefly to this PR, and while I ended up not needing it, it appears that this changes the default behavior to determine deps by source file analysis. I want to leave this here so this isn't a surprise in case we end up introducing other nx plugins later on!